src/sessions/PageManager.ts

```diff
@@ -188,9 +188,17 @@ export class PageManager {
         }
 
         if (customPageAttributes) {
-            Object.keys(customPageAttributes).forEach((attribute) => {
-                this.attributes[attribute] = customPageAttributes[attribute];
-            });
+            if (customPageAttributes.pageTags) {
+                this.attributes.pageTags = customPageAttributes.pageTags;
+            }
+            if (customPageAttributes.pageAttributes) {
+                Object.keys(customPageAttributes.pageAttributes).forEach(
+                    (attribute) => {
+                        this.attributes[attribute] =
+                            customPageAttributes.pageAttributes[attribute];
+                    }
+                );
+            }
```

Comment:
```suggestion
```suggestion
        if (customPageAttributes?.pageTags) {
            this.attributes.pageTags = customPageAttributes.pageTags;
        }
        if (customPageAttributes?.pageAttributes) {
            Object.keys(customPageAttributes.pageAttributes).forEach(
                (attribute) => {
                    this.attributes[attribute] =
                        customPageAttributes.pageAttributes[attribute];
                }
            );
        }
```
```

Reply:
Refactored accordingly